### PR TITLE
Hunter entity introduced

### PR DESCRIPTION
**Pull request by Buvko**
Hunter entity introduced
+ learning to work with forks and pull requests
+ fixed issue with two persistence modules

### CODE REVIEW

File: persistence/src/main/java/cz/muni/fi/pa165/mushrooms/entity/Hunter.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package cz.muni.fi.pa165.mushrooms.entity;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Created by Matúš on 18.10.2017.
```
Comment (bkompis) [inline persistence/src/main/java/cz/muni/fi/pa165/mushrooms/entity/Hunter.java]: Use your github account name here

File: persistence/src/main/java/cz/muni/fi/pa165/mushrooms/entity/Hunter.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
+package cz.muni.fi.pa165.mushrooms.entity;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Created by Matúš on 18.10.2017.
+ */
+@Entity
+public class Hunter {
```
Comment (bkompis) [inline persistence/src/main/java/cz/muni/fi/pa165/mushrooms/entity/Hunter.java]: I'd call the entity 'MushroomHunter', to match the issue and diagrams.

File: persistence/src/main/java/cz/muni/fi/pa165/mushrooms/entity/Hunter.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    @GeneratedValue(strategy= GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull
+    @Column(nullable=false,unique=true)
+    private String firstName;
+
+    @NotNull
+    @Column(nullable=false,unique=true)
+    private String surName;
```
Comment (bkompis) [inline persistence/src/main/java/cz/muni/fi/pa165/mushrooms/entity/Hunter.java]: surName should be surname (it's one word)

File: persistence/src/main/java/cz/muni/fi/pa165/mushrooms/entity/Hunter.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    public Long getId() {
+        return id;
+    }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public String getSurName() {
```
Comment (bkompis) [inline persistence/src/main/java/cz/muni/fi/pa165/mushrooms/entity/Hunter.java]: rename to match the attribute

File: persistence/src/main/java/cz/muni/fi/pa165/mushrooms/entity/Hunter.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    public String getPersonalInfo() {
+        return personalInfo;
+    }
+
+    public void setFirstName(String firstName) {
+        this.firstName = firstName;
+    }
+
+    public void setSurName(String surName) {
```
Comment (bkompis) [inline persistence/src/main/java/cz/muni/fi/pa165/mushrooms/entity/Hunter.java]: rename to match the attribute

File: persistence/src/main/java/cz/muni/fi/pa165/mushrooms/entity/Hunter.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    @Override
+    public int hashCode() {
+        int result = getFirstName().hashCode();
+        result = 31 * result + getSurName().hashCode();
+        result = 31 * result + (getPersonalInfo() != null ? getPersonalInfo().hashCode() : 0);
+        return result;
+    }
+}
```
Comment (bkompis) [inline persistence/src/main/java/cz/muni/fi/pa165/mushrooms/entity/Hunter.java]: If we want to add an 'admin' to the system, the simplest way would be to add a boolean flag to this entity (isAdmin), which can be used further.

**Review by bkompis — CHANGES_REQUESTED** [review]
Good first draft, some renames should be made. Based on the team decision, we might want to add one more attribute (isAdmin flag) to the entity.

**Review by bkompis — APPROVED** [review]
All fixed, I approve.

**Comment by bencikpeter** [comment]
Minor detail, 
` @NotNull
    @Column(nullable=false,unique=true)
    private String firstName;`

I believe first name should not be unique...

**Pull request merged**